app/models/miq_server/worker_management/monitor/kubernetes.rb

```diff
@@ -15,31 +20,89 @@ def failed_deployments(restart_count = 5)
     current_pods.values.select { |h| h[:last_state_terminated] && h.fetch(:container_restarts, 0) > restart_count }.collect { |h| h[:label_name] }
   end
 
+  def sync_deployment_settings
+    checked_deployments = Set.new
+    podified_miq_workers.each do |worker|
+      next if checked_deployments.include?(worker.worker_deployment_name)
+
+      if deployment_resource_constraints_changed?(worker)
+        _log.info("Constraints changed, patching deployment: [#{worker.worker_deployment_name}]")
+        worker.patch_deployment
+      end
+      checked_deployments << worker.worker_deployment_name
+    end
+  end
+
+  def podified_miq_workers
+    # Cockpit is a threaded worker in the orchestrator that spins off a process it monitors and isn't a pod worker.
+    miq_workers.where.not(:type => %w[MiqCockpitWsWorker])
+  end
+
+  def deployment_resource_constraints_changed?(worker)
+    return false unless Settings.server.worker_monitor.enforce_resource_constraints
+
+    container = current_deployments.fetch_path(worker.worker_deployment_name, :spec, :template, :spec, :containers).try(:first)
+    current_constraints = container.try(:fetch, :resources, nil) || {}
+    desired_constraints = worker.resource_constraints
+    constraints_changed?(current_constraints, desired_constraints)
+  end
+
+  def constraints_changed?(current, desired)
+    if current.present? && desired.present?
+      !cpu_value_eql?(current.fetch_path(:requests, :cpu), desired.fetch_path(:requests, :cpu)) ||
+        !cpu_value_eql?(current.fetch_path(:limits, :cpu), desired.fetch_path(:limits, :cpu)) ||
+        !mem_value_eql?(current.fetch_path(:requests, :memory), desired.fetch_path(:requests, :memory)) ||
+        !mem_value_eql?(current.fetch_path(:limits, :memory), desired.fetch_path(:limits, :memory))
+    else
+      # current, no desired    => changed
+      # no current, desired    => changed
+      # no current, no desired => unchanged
+      current.blank? ^ desired.blank?
+    end
+  end
+
   private
 
-  def start_pod_monitor
+  def cpu_value_eql?(current, desired)
```

Comment:
What are the possible incoming values?  Could you get a `0.5` a.k.a 50%?

Reply:
Good eyes... yes, 0.5 is allowed.  "A request with a decimal point, like 0.1, is converted to 100m by the API, and precision finer than 1m is not allowed", from: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#requests-and-limits

Pushed changes to use `to_f` to detect "0.5" == "500m"